portal/src/graphql/portal/AuthenticationCountryCallingCodeList.tsx

```diff
@@ -0,0 +1,490 @@
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import Fuse from "fuse.js";
+import cn from "classnames";
+import produce from "immer";
+import {
+  Checkbox,
+  DetailsList,
+  IColumn,
+  IObjectWithKey,
+  SelectionMode,
+  ICheckboxProps,
+  IDetailsHeaderProps,
+  IRenderFunction,
+  DetailsHeader,
+  DetailsRow,
+  IDetailsRowProps,
+  SelectAllVisibility,
+  SearchBox,
+} from "@fluentui/react";
+import { Context, FormattedMessage } from "@oursky/react-messageformat";
+
+import { useTextField } from "../../hook/useInput";
+import { OrderColumnButtons, swap } from "../../DetailsListWithOrdering";
+import { useGetTelecomCountryName } from "../../util/translations";
+import countryCallingCodeMap from "../../data/countryCodeMap.json";
+
+import styles from "./AuthenticationCountryCallingCodeList.module.scss";
+
+interface CountryCallingCodeListProps {
+  allCountryCallingCodes: string[];
+  selectedCountryCallingCodes: string[];
+  onSelectedCountryCallingCodesChange: (newSelectedCodes: string[]) => void;
+}
+
+interface CountryCallingCodeListItem extends IObjectWithKey {
+  key: string;
+  selected: boolean;
+  countryName: string;
+  callingCode: string;
+}
+
+type CountryCallingCodeListData = Record<
+  string,
+  { key: string; countryName: string; callingCode: string }
+>;
+
+interface CountryCallingCodeListItemCheckboxProps extends ICheckboxProps {
+  index?: number;
+  onCheckboxClicked: (index: number, checked: boolean) => void;
+}
+
+interface CountryCallingCodeListSelectAllProps extends ICheckboxProps {
+  isPartiallySelected: boolean;
+  isAllSelected: boolean;
+  selectAll: () => void;
+  unselectAll: () => void;
+}
+
+const HEADER_STYLE = {
+  check: {
+    width: "35px !important",
+    opacity: "1 !important",
+    paddingLeft: "15px !important",
+  },
+};
+
+const SEARCH_BOX_STYLE = {
```

Comment:
Why use custom styles instead of setting it in `searchBox` CSS class?

Reply:
I followed the example in Fluent UI doc, I think this can be achieved with CSS class